examples/common.cpp

```diff
@@ -281,7 +285,14 @@ gpt_vocab::id gpt_sample_top_k_top_p(
     {
         const double scale = 1.0/temp;
         for (int i = 0; i < n_logits; ++i) {
-            logits_id.push_back(std::make_pair(logits[i]*scale, i));
+            // repetition penalty from ctrl paper (https://arxiv.org/abs/1909.05858)
+            // credit https://github.com/facebookresearch/llama/compare/main...shawwn:llama:main
+            // if score < 0 then repetition penalty has to multiplied to reduce the previous token probability
+            if (logits[i] < 0.0f) {
+                logits_id.push_back(std::make_pair(logits[i]*scale*repetition_penalty, i));
+            } else {
+                logits_id.push_back(std::make_pair(logits[i]*scale/repetition_penalty, i));
+            }
```

Comment:
Hm, I think we want to check if the token is repeated first?

Reply:
I think I missed an outermost loop. Checking.